processing/src/main/java/org/apache/druid/segment/nested/ScalarDoubleColumnAndIndexSupplier.java

```diff
@@ -242,12 +256,24 @@ public double estimateSelectivity(int totalRows)
         public <T> T computeBitmapResult(BitmapResultFactory<T> bitmapResultFactory)
         {
           if (doubleValue == null) {
-            if (inputNull) {
+            if (inputNull && NullHandling.sqlCompatible()) {
```

Comment:
Should be consistent between using a field (this could be `inputNull && !replaceWithDefault`) or using `NullHandling`. I don't see a good reason to mix them.